azurerm/resource_arm_virtual_network_gateway_connection.go

```diff
@@ -194,9 +305,26 @@ func resourceArmVirtualNetworkGatewayConnectionRead(d *schema.ResourceData, meta
 		d.Set("local_network_gateway_id", conn.LocalNetworkGateway2.ID)
 	}
 
-	d.Set("enable_bgp", conn.EnableBgp)
-	d.Set("routing_weight", conn.RoutingWeight)
-	d.Set("shared_key", conn.SharedKey)
+	if conn.EnableBgp != nil {
+		d.Set("enable_bgp", conn.EnableBgp)
+	}
+
+	if conn.UsePolicyBasedTrafficSelectors != nil {
+		d.Set("use_policy_based_traffic_selectors", conn.UsePolicyBasedTrafficSelectors)
+	}
+
+	if conn.RoutingWeight != nil {
+		d.Set("routing_weight", conn.RoutingWeight)
+	}
+
+	if conn.SharedKey != nil {
+		d.Set("shared_key", conn.SharedKey)
+	}
+
+	if conn.IpsecPolicies != nil {
+		ipsec_policies := flattenArmVirtualNetworkGatewayConnectionIpsecPolicies(conn.IpsecPolicies)
+		d.Set("ipsec_policy", ipsec_policies)
```

Comment:
given this is a complex object, can we check for errors here via:

```
if err := d.Set("ipsec_policy", ipsec_policies); err != nil {
  return fmt.Errorf("Error setting `ipsec_policy`: %+v", err)
}
```

_minor_ could we also rename the variable `ipsec_policies` to `ipsecPolicies` to match the other variables?

Reply:
Regarding the variable naming, I found it helpful to quickly identify the source of which the value originates. I.e. I chose CamelCase for values from the Azure Go SDK whereas snake_case for value from the Terraform schema.

However, to match the other resources, I will revert it.